src/main/webapp/app/modeling-assessment-editor/modeling-assessment-conflict/modeling-assessment-conflict.component.ts

```diff
@@ -58,13 +173,56 @@ export class ModelingAssessmentConflictComponent implements OnInit, AfterViewIni
 
     private updateSelectedConflict() {
         this.currentConflict = this.conflicts[this.conflictIndex];
-        this.conflictingResult = this.currentConflict.conflictingResults[0];
+        this.conflictingResult = this.currentConflict.resultsInConflict[0];
         this.conflictingModel = JSON.parse((this.conflictingResult.result.submission as ModelingSubmission).model);
         this.updateHighlightedElements();
+        this.updateHighlightColor();
     }
 
     private updateHighlightedElements() {
-        this.modelHighlightedElementIds = new Set<string>([this.currentConflict.modelElementId]);
+        this.modelHighlightedElementIds = new Set<string>([this.currentConflict.causingConflictingResult.modelElementId]);
         this.conflictingModelHighlightedElementIds = new Set<string>([this.conflictingResult.modelElementId]);
     }
+
+    private updateCurrentState() {
+        const mergedFeedback = this.mergedFeedbacks.find((feedback: Feedback) => feedback.referenceId === this.currentConflict.causingConflictingResult.modelElementId);
+        const conflictingFeedback = this.conflictingResult.result.feedbacks.find((feedback: Feedback) => feedback.referenceId === this.conflictingResult.modelElementId);
+        if (mergedFeedback.credits !== conflictingFeedback.credits) {
+            this.conflictResolutionStates[this.conflictIndex] = ConflictResolutionState.ESCALATED;
+        } else {
+            this.conflictResolutionStates[this.conflictIndex] = ConflictResolutionState.RESOLVED;
+        }
+        this.updateHighlightColor();
+        this.updateOverallResolutioState();
+    }
+
+    private updateHighlightColor() {
+        switch (this.conflictResolutionStates[this.conflictIndex]) {
+            case ConflictResolutionState.UNHANDLED:
+                // this.highlightColor = 'rgba(219, 53, 69, 0.6)';
+                this.highlightColor = 'rgba(0, 123, 255, 0.6)';
+                break;
+            case ConflictResolutionState.ESCALATED:
+                this.highlightColor = 'rgba(255, 193, 7, 0.6)';
+                break;
+            case ConflictResolutionState.RESOLVED:
+                this.highlightColor = 'rgba(40, 167, 69, 0.6)';
+                break;
+        }
+    }
+
+    private updateOverallResolutioState() {
+        for (const state of this.conflictResolutionStates) {
```

Comment:
This could also very easily be expressed in a functional way:
this.conflictsAllHandled = this.conflictResolutionStates.every(state => state !== ConflictResolutionState.UNHANDLED);

Reply:
Well thats quite a useful method to know.. thanks for the hint. Changed the method accordingly